src/tests/e2etest/HostNameTests.cs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Azure.Devices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace e2etesting
+{
+    [TestFixture]
+    public class HostNameTests : E2eTest
+    {   
+        string ComponentName = "HostName";
+        public partial class ExpectedDesiredProperty
+        {
+            public string value { get; set; }
+            public int ac { get; set; }
```

Comment:
why not `JsonIgnore` this property? This would make the `ExpectedDesiredProperty` much easier/cleaner to read when defining the property.

Reply:
ok, i can see the value of checking for equality on `ac`, but i don't think it should be a property exposed for the test to define.
I would suggest adding/checking these properties in the `E2eTest::AreEqualByJson` instead of hard coding them here. Its confusing to read for a module developer who is using this as a reference.